### PR TITLE
Repo metadata updates

### PR DESCRIPTION
**Pull request by andrewschoen**
1) adds project name to repo metadata and search filters docs

2) adds distro_codename to repo metadata

### CODE REVIEW

File: shaman/models/repos.py [hunk 1]
```diff
@@ -52,16 +52,23 @@ def __repr__(self):
             return '<Repo detached>'
 
     def __json__(self):
+        from shaman.util import parse_distro_release
```
Comment (alfredodeza) [inline shaman/models/repos.py]: why is this import here?

File: shaman/models/repos.py [hunk 2]
```diff
@@ -52,16 +52,23 @@ def __repr__(self):
...
+        codename, version = parse_distro_release(
+            self.distro_version,
```
Comment (alfredodeza) [inline shaman/models/repos.py]: Does this need to use `self.`?

File: shaman/models/repos.py (hunk 1 above)
Comment (andrewschoen) [inline shaman/models/repos.py]: circular import problems. ``shaman.util`` imports ``shaman.models``. putting this here was a quick fix.

File: shaman/models/repos.py (hunk 2 above)
Comment (andrewschoen) [inline shaman/models/repos.py]: Yes, I'm not sure how else I could access the ``distro_version``.
